src/controllers/secret.go

```diff
@@ -0,0 +1,194 @@
+	}
+
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      request.Name,
+			Namespace: request.Namespace,
+		},
+		Type: secretType,
+		Data: data,
+	}
+
+	newSecret, err := n.client.CoreV1().Secrets(namespace).Create(n.ctx, secret, metav1.CreateOptions{})
+	if err != nil {
+		n.logger.Error(fmt.Sprintf("Could not create secret %q with error: %s", name, err.Error()))
+		return response, err
+	}
+
+	n.logger.Debug(fmt.Sprintf("Created secret %q successfully", name))
+
+	response.Type = string(newSecret.Type)
+	response.Name = newSecret.Name
+	response.Namespace = newSecret.Namespace
+	return response, err
+}
+
+func (n *secretController) GetSecrets(request types.GetSecretsRequest) (types.GetSecretsResponse, error) {
+	n.logger.Debug(fmt.Sprintf("Trying to get all secrets in %s namespace", request.Namespace))
+
+	response := types.GetSecretsResponse{}
+	namespace := request.Namespace
+	secrets, err := n.client.CoreV1().Secrets(namespace).List(n.ctx, metav1.ListOptions{})
+	if err != nil {
+		n.logger.Error(fmt.Sprintf("Could not get secrets with error: %s", err.Error()))
+		return response, err
+	}
+
+	n.logger.Debug("Fetched all secrets successfully")
+
+	response.Count = len(secrets.Items)
+	for _, secret := range secrets.Items {
+		response.Secrets = append(
+			response.Secrets,
+			types.Secret{
+				Type:      string(secret.Type),
+				Name:      secret.Name,
+				Namespace: secret.Namespace,
+			})
+	}
+	return response, nil
+}
+
+func (n *secretController) GetSecret(request types.GetSecretRequest) (types.GetSecretResponse, error) {
+	n.logger.Debug(fmt.Sprintf("Trying to get all secrets in %s namespace", request.Namespace))
+
+	response := types.GetSecretResponse{}
+	name := request.Name
+	namespace := request.Namespace
+
+	secret, err := n.client.CoreV1().Secrets(namespace).Get(n.ctx, name, metav1.GetOptions{})
+	if err != nil {
+		n.logger.Error(fmt.Sprintf("Could not get secret %s with error: %s", name, err.Error()))
+		return response, err
+	}
+
+	n.logger.Debug("Fetched secret successfully")
+
+	response.Id = string(secret.GetUID())
+	response.Name = secret.Name
+	response.Type = string(secret.Type)
+	for k, v := range secret.Data {
+		response.Data = append(response.Data, types.KeyValue{
+			Key:   k,
+			Value: string(v),
+		})
+	}
+	return response, nil
+}
+
+func (n *secretController) PatcnSecret(request types.UpdateSecretRequest) (types.UpdateSecretResponse, error) {
+	n.logger.Debug(fmt.Sprintf("Trying to patch an existing secret in %s namespace", request.Namespace))
+
+	response := types.UpdateSecretResponse{}
+	name := request.Name
+	namespace := request.Namespace
+	secret := v1.Secret{
+		Data: make(map[string][]byte),
+	}
+
+	for _, kv := range request.Data {
+		secret.Data[kv.Key] = []byte(kv.Value)
+	}
+
+	data, err := json.Marshal(secret)
+	if err != nil {
+		n.logger.Error(fmt.Sprintf("Could not marshal secret %s with error: %s", name, err.Error()))
```

Comment:
Also here, use `%q` for names (so that they would appear in quotation marks in the logs) and `%v` for the errors.